src/directLine.ts

```diff
@@ -272,15 +277,31 @@ export class DirectLine implements IBotConnection {
     constructor(options: DirectLineOptions) {
         this.secret = options.secret;
         this.token = options.secret || options.token;
+        const useWebSockets:boolean = options.webSocket && typeof WebSocket !== 'undefined' && WebSocket !== undefined;
+
         if (options.domain)
             this.domain = options.domain;
         if (options.webSocket !== undefined)
             this.webSocket = options.webSocket;
-
+        if (options.conversationId !== undefined) {
+            this.conversationId = options.conversationId;
+            if (options.watermark !== undefined) {
```

Comment:
Either take this case out from under the check for `conversationId` or put the `streamUrl` check next to it so that you're treating them the same. I slightly prefer taking this out, because I like giving a warning if `conversationId` was not supplied.